doc/joss/paper.md

```diff
@@ -0,0 +1,141 @@
+  - name: Fritz G{\"o}bel
+    affiliation: 1 
+  - name: Thomas Gr{\"u}tzmacher 
+    affiliation: 1
+  - name: Pratik Nayak 
+    orcid: 0000-0002-7961-1159
+    affiliation: 1 
+  - name: Tobias Ribizel 
+    affiliation: 1 
+  - name: Yu-Hsiang Tsai 
+    orcid: 0000-0001-5229-3739
+    affiliation: 1 
+affiliations:
+ - name: Karlsruhe Institute of Technology
+   index: 1
+ - name: Innovative Computing Laboratory, University of Tennessee, Knoxville 
+   index: 2
+ - name: University of Jaume I 
+   index: 3
+ - name: The University of Tokyo 
+   index: 4
+date: 6th March, 2020.
+bibliography: paper.bib
+
+---
+
+# Summary
+
+Ginkgo is a production-ready sparse linear algebra library for high performance 
+computing on GPU-centric architectures with a high level of performance portability
+and software sustainability.
+
+The library focuses on solving sparse linear systems and accommodates a large variety
+of matrix formats, state-of-the-art iterative (Krylov) solvers and preconditioners, 
+which make the library suitable for a variety of scientific applications. Ginkgo
+supports many architectures such as multi-threaded CPU, NVIDIA GPU's, and AMD GPU's.
+The heavy use of modern C++ 11 features such as dynamic polymorphism and lambdas
+simplifies the addition of new executor paradigms and algorithmic functionality
+without introducing performance degradations. 
+
+Ginkgo is also a part of the xSDK effort [@xsdk] which aims to provide infrastructure
+for and interoperability of a collection of related and complementary software 
+elements to foster rapid and efficient development of scientific applications
+using High Performance Computing. Within this effort, we support interoperability with
+application libraries such as `deal.ii`[@dealii] and `mfem`[@mfem]. Ginkgo provides
+wrappers within these two libraries so that they can take advantage of the features of
+Ginkgo. 
+
+# Features
+
+As sparse linear algebra is one of the main focus of Ginkgo, we provide a variety of 
+sparse matrix formats such as COO, CSR, ELL, HYBRID and SELLP along with highly tuned
+Sparse Matrix Vector product (SpMV) kernels. The SpMV kernel is a key building 
+blocks of vitually all iterative solvers and typically accounts for a significant 
+fraction of the application runtime. Additionally, we also provide vhigh performance 
+conversion routines between the different formats enhancing their flexibility.
+
+Ginkgo provides multiple iterative solvers such as the Krylov subspace
+methods: Conjugate gradient (CG), Flexible Conjugate Gradient (FCG), Bi-Conjugate
+Gradient (BiCG) and its stabilized version (Bi-CGSTAB), Generalized Minimal
+residual method (GMRES) and more generic methods such as Iterative Refinement,
+which forms the basis of many relaxation methods. Ginkgo also features support for 
+direct and iterative triangular solves within incomplete factorization preconditioners.
+
+Ginkgo features some of the best preconditioners such as the Block Jacobi
```

Comment:
maybe add *general-purpose*? Other preconditioners might arguably be better for specific use cases.